src/node_context_data.h

```diff
@@ -47,14 +47,84 @@ namespace node {
 #define NODE_CONTEXT_TAG 39
 #endif
 
+/**
+ * These are indexes used for to set Node.js-specific data in the V8 contexts
+ * via v8::Context::SetAlignedPointerInEmbedderData() (for pointers) and
+ * v8::Context::SetEmbedderData() (for v8::Values).
+ *
+ * There are five types of contexts in Node.js:
+ * 1. Default V8 context, with nothing Node.js-specific. This is normally only
+ *    created by the embedders that uses v8::Context APIs directly and is not
+ *    available through the Node.js JS APIs. Its context snapshot in the
+ *    built-in v8 startup snapshot is stored using
+ *    v8::SnapshotCreator::SetDefaultContext() - effectively at index 0.
+ * 2. Default Node.js main context. When Node.js is launched typically there is
+ *    a main thread on which a main node::Environment is created. The
+ *    Environment is associated with its own v8::Isolate (env->isolate()) and a
+ *    main v8::Context (env->context()). The corresponding data structure is
+ *    node::NodeMainInstance and the Environment is created in
+ *    node::NodeMainInstance::Run(). Its context snapshot in the built-in V8
+ *    startup snapshot is stored at node::SnapshotData::kNodeMainContextIndex.
+ * 3. Node.js worker context. When a Worker instance is created via
+ *    new worker_threads.Worker(), it gets its own OS thread, its
+ *    own node::Environment, its own v8::Isolate and its own v8::Context.
+ *    The corresponding data structure is node::Worker and the Environment
+ *    is created in node::NodeMainInstance::Run().
```

Comment:
A worker thread can be created in a worker thread as well.

```suggestion
 *    The corresponding data structure is node::Worker.
```

Reply:
Oh actually I think it's copy-pasta and I meant `..is created node::Worker::Run()`..